src/ls.js

```diff
@@ -36,91 +37,24 @@ function _ls(options, paths) {
     options.all = true;
   }
 
-  if (!paths)
-    paths = ['.'];
-  else if (typeof paths === 'object')
-    paths = paths; // assume array
-  else if (typeof paths === 'string')
-    paths = [].slice.call(arguments, 1);
+  if (typeof paths === 'string') paths = [].slice.call(arguments, 1);
+  paths = paths || [];
 
-  var list = [];
-
-  // Conditionally pushes file to list - returns true if pushed, false otherwise
-  // (e.g. prevents hidden files to be included unless explicitly told so)
-  function pushFile(file, query) {
-    // hidden file?
-    if (path.basename(file)[0] === '.') {
-      // not explicitly asking for hidden files?
-      if (!options.all && !(path.basename(query)[0] === '.' && path.basename(query).length > 1))
-        return false;
-    }
+  paths = paths.map(function (path) {
+    path += (options.recursive ? '/**/*.*' : '/*.*');
+    return path.replace(/\\/g, '/');
+  });
 
-    if (common.platform === 'win')
-      file = file.replace(/\\/g, '/');
+  var files = paths.reduce(function (files, path) {
```

Comment:
I think if we change this to `files = common.expand(files, { dot: options.all })`, and update `expand()` to take glob's options as a second parameter, this will eliminate the dependency on glob here.